python/lsst/daf/butler/formatters/logs.py

```diff
@@ -0,0 +1,121 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("ButlerLogRecordsFormatter", )
+
+from typing import (
+    Any,
+    Optional,
+    Type,
+)
+
+from lsst.daf.butler import ButlerLogRecords, ButlerLogRecord
+from .json import JsonFormatter
+
+
+class ButlerLogRecordsFormatter(JsonFormatter):
+    """Read and write log records in JSON format.
+
+    This is a naive implementation that treats everything as a pydantic.
+    model.  In the future this may be changed to be able to read
+    ``ButlerLogRecord`` one at time from the file and return a subset
+    of records given some filtering parameters.
+    """
+
+    def _readFile(self, path: str, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read a file from the path in JSON format.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open JSON format file.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Data as Python object read from JSON file.
+
+        Notes
+        -----
+        Can read two forms of JSON log file. It can read a full JSON
+        document created from `ButlerLogRecords`, or a stream off standalone
+        JSON documents with a log record per line.
+        """
+        if pytype is None:
+            pytype = ButlerLogRecords
+        elif not issubclass(pytype, ButlerLogRecords):
+            raise RuntimeError(f"Python type {pytype} does not seem to be a ButlerLogRecords type")
+
+        return pytype.from_file(path)
+
+    def _fromBytes(self, serializedDataset: bytes, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        serializedDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        inMemoryDataset : `object`
+            The requested data as a Python object or None if the string could
+            not be read.
+        """
+        # Duplicates some of the logic from ButlerLogRecords.from_file
+        if pytype is None:
+            pytype = ButlerLogRecords
+        elif not issubclass(pytype, ButlerLogRecords):
+            raise RuntimeError(f"Python type {pytype} does not seem to be a ButlerLogRecords type")
+
+        if serializedDataset.startswith(b"["):
+            return pytype.parse_raw(serializedDataset)
+
+        if not serializedDataset.startswith(b"{"):
+            raise RuntimeError(f"These bytes do not look like JSON -- starts with '{serializedDataset[0]}'")
```

Comment:
I think this will fail with different exception if `serializedDataset` is empty

Reply:
Good point. I've been wondering whether to also move this to `ButlerLogRecords` as `.from_raw()` since at the moment the formatter has the byte reading version but the class itself has the file reading version.